pydantic/class_validators.py

```diff
@@ -165,14 +169,14 @@ def get_validators(self, name: str) -> Optional[Dict[str, Validator]]:
         if name != ROOT_KEY:
             validators += self.validators.get('*', [])
         if validators:
-            return {v.func.__name__: v for v in validators}
+            return {getattr(v.func, '__name__', '<default name>'): v for v in validators}
```

Comment:
Again, I'd try to return something that would be a bit more unique, using id seems reasonable to me if there is no `__name__`.